providers/caching/redis.js

```diff
@@ -3,31 +3,48 @@
 
 const redis = require('redis')
 const util = require('util')
+const pako = require('pako')
 
 const objectPrefix = '*!~%'
 
 class RedisCache {
   constructor(options) {
-    this.redis = redis.createClient(options)
-    this._redisGet = util.promisify(this.redis.get)
-    this._redisSet = util.promisify(this.redis.set)
+    this.options = options
+  }
+
+  initialize() {
+    this.redis = redis.createClient(6380, this.options.service, {
+      auth_pass: this.options.apiKey,
+      tls: { servername: this.options.service }
+    })
+    this._redisGet = util.promisify(this.redis.get).bind(this.redis)
+    this._redisSet = util.promisify(this.redis.set).bind(this.redis)
+    this._redisDel = util.promisify(this.redis.del).bind(this.redis)
   }
 
   async get(item) {
     const cacheItem = await this._redisGet(item)
-    if (cacheItem && cacheItem.startsWith(objectPrefix)) {
+    if (!cacheItem) return null
+    const result = pako.inflate(cacheItem, { to: 'string' })
+    if (result.startsWith(objectPrefix)) {
       try {
-        return JSON.parse(cacheItem.substring(4))
+        return JSON.parse(result.substring(4))
       } catch (error) {
         return null
       }
     }
-    return cacheItem
+    return result
   }
 
   async set(item, value, expirationSeconds) {
     if (typeof value !== 'string') value = objectPrefix + JSON.stringify(value)
-    await this._redisSet(item, value, 'EX', expirationSeconds)
+    const data = pako.deflate(value, { to: 'string' })
+    if (expirationSeconds) await this._redisSet(item, data, 'EX', expirationSeconds)
```

Comment:
with no expiration will they just monotonically fill up until the server stop or is there a more global MRU policy or some such?

Reply:
The redis itself has a maxmemory policy to evict keys when there is pressure.